### PR TITLE
[mypyc] Translate attribute access to index access for named tuples

### PR DESCRIPTION
**Pull request by ilevkivskyi**
Fixes https://github.com/mypyc/mypyc/issues/638

Previously `PyObject_GetAttr` was used for named tuples, which is slower.

### CODE REVIEW

File: mypyc/test-data/irbuild-tuple.test [hunk 1]
```diff
@@ -162,3 +162,35 @@ L4:
     r8 = None
     return r8
 
+[case testNamedTupleAttribute]
+from typing import NamedTuple
+
+NT = NamedTuple('NT', [('x', int), ('y', int)])
```
Comment (JelleZijlstra) [inline mypyc/test-data/irbuild-tuple.test:168]: Might be worth adding a unit test for a NamedTuple that inherits from another, something like the example from #8663.

**Comment by ilevkivskyi** [comment]
@msullivan 
> I agree with Jelle's suggestion

I was trying to add a subclass in compiled code in different ways, but I was always getting `Inheriting from most builtin types is unimplemented`. I therefore only added a test with a subclass in interpreted code.

While doing this, I actually noticed that using index access is unsafe. Imagine someone overrides the accessor in an interpreted subclass:
```
class NT(NamedTuple):
    x: int
    y: int

def get_y(nt: NT) -> int:
    return nt.y

[file driver.py]
from native import NT, get_y

class C(NT):
    @property
    def y(self):
        return 2 * self[1]
assert get_y(C(1, 1)) == 2  # Works in Python but fails if we compile code
```
Should we worry about such overrides? Or people who write such overrides deserve this?

**Comment by ilevkivskyi** [comment]
Also the perf win from this is pretty minor, around 1% for mypy self-check.

**Comment by msullivan** [comment]
We have the same problem for attribute access and property overrides for all inherited subclasses of compiled classes, as it turns out, and we don't worry about it too much there...

**Comment by ilevkivskyi** [comment]
> We have the same problem for attribute access and property overrides for all inherited subclasses of compiled classes, as it turns out, and we don't worry about it too much there...

OK, I will merge this one then.

**Pull request merged**